### PR TITLE
feat(*): support to register all methods for a Definition

### LINKED ISSUE #343: Can support ALL option for HTTP Method?
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
> /kind feature

**What happened**:
now only support  Method  single, for example:
```
Definitions: []definition.Definition{
     {
	    Method:  definition.Delete,
	    Handler: xxxxx,
     },
},
```
**What you expected to happen**:
I want to support ALL for HTTP Method when reverseproxy, for example
```
Definitions: []definition.Definition{
     {
	    Method:  definition.ALL,
	    Handler: xxxxx,
     },
},
```

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by iawia002**
**What this PR does / why we need it**:

 close #343 

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @gaocegege @simon-cj @hezhizhen 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: utils/api/definitions.go [hunk 1]
```diff
@@ -111,6 +112,10 @@ func NewDefinition(tc *TypeContainer, d *definition.Definition) (*Definition, er
 		ErrorProduces: d.ErrorProduces,
 		Function:      tc.NameOfInstance(d.Function),
 	}
+	if d.Method == definition.Any {
+		cd.HTTPMethod = string(definition.Any)
```
Comment (hezhizhen) [inline utils/api/definitions.go:116]: 这里为啥不是在 service 的方法中增加对 any 的处理？

Comment (iawia002) [reply utils/api/definitions.go:116]: 这里是 API 文档生成那儿要单独处理下 any 的情况

**Pull request merged**